src/main/java/Assignment1.java

```diff
@@ -0,0 +1,175 @@
+        if (input_string.split("-name", -1).length - 1 != 1) {
+            System.out.println("Invalid name");
+            return false;
+        }
+
+        // input can have only 1 price argument
+        if (input_string.split("-price", -1).length - 1 != 1) {
+            System.out.println("Invalid price");
+            return false;
+        }
+
+        // input can have only 1 quantity argument
+        if (input_string.split("-quantity", -1).length - 1 != 1) {
+            System.out.println("Invalid quantity");
+            return false;
+        }
+
+        // input can have only 1 type argument
+        if (input_string.split("-type", -1).length - 1 != 1) {
+            System.out.println("Invalid type");
+            return false;
+        }
+
+        // input should begin with name argument
+        if (input_string.indexOf("-name") != 0) {
+            System.out.println("Invalid name postion");
+            return false;
+        }
+
+        String name = extractWord(input_string, input_string.indexOf("-name") + 5).trim();
+        String quantity = extractWord(input_string, input_string.indexOf("-quantity") + 9).trim();
+        String price = extractWord(input_string, input_string.indexOf("-price") + 6).trim();
+        String type = extractWord(input_string, input_string.indexOf("-type") + 5).trim();
+
+        // check empty string
+        if (name.isEmpty()) {
+            System.out.println("Invalid name");
+            return false;
+        }
+        if (price.isEmpty()) {
+            System.out.println("Invalid price");
+            return false;
+        }
+        if (quantity.isEmpty()) {
+            System.out.println("Invalid quantity");
+            return false;
+        }
+        if (type.isEmpty()) {
+            System.out.println("Invalid type");
+            return false;
+        }
+        // name should only have alphabets.
+        if (!name.matches("^[ A-Za-z]+$")) {
+            System.out.println("Invalid name");
+            return false;
+        }
+        // quantity should only have numbers.
+        if (!quantity.matches("[0-9]*") || Integer.parseInt(quantity) <= 0) {
+            System.out.println("Invalid quantity");
+            return false;
+        }
+        // price should only have numbers and can have decimal (.) once.
+        if (!price.matches("[0-9.]*") || (price.length() - price.replace(".", "").length() > 1)) {
+            System.out.println("Invalid price");
+            return false;
+        }
+        // type check.
+        if (!type.equals("raw") && !type.equals("manufactured") && !type.equals("imported")) {
+            System.out.println("Invalid type");
+            return false;
+        }
+        return true;
+    }
+
+    // method to extract data from input to member variables
+    void extractData(String input_string) {
+        String name = extractWord(input_string, input_string.indexOf("-name") + 5).trim();
```

Comment:
create constants for -name, -quantity, -price, -type as these strings are used multiple times in the business logic as well.